### PR TITLE
[PROJECT] Clockwork Marauder

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
The first (technically second) big-ticket scripture you can get. 2000W, 6 cogs, and 100 vitality can get you a clockwork marauder. They have 140 HP, dealing 24 damage to mobs. They also have a shield, which is able to take 4 hits before breaking (all damage taken by the shield is absorbed). If you become too damaged, ask a cultist to use a welder to repair your health and shield.

## How This Contributes To The Skyrat Roleplay Experience
Vitality now has a use

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

### CODE REVIEW

File: modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	GLOB.clockwork_marauders -= src
+	return ..()
+
+
+/mob/living/basic/clockwork_marauder/attacked_by(obj/item/attacking_item, mob/living/user)
+	if(istype(attacking_item, /obj/item/nullrod))
+		apply_damage(15, BURN)
+
+		if(shield_health)
+			damage_shield()
```
Comment (MrMelbert) [inline modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm]: `/datum/element/bane` and `COMSIG_LIVING_BANED` are crying right now

File: modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+/mob/living/basic/clockwork_marauder/proc/damage_shield()
+	shield_health--
+	playsound(src, 'sound/magic/clockwork/anima_fragment_attack.ogg', 60, TRUE)
+	if(!shield_health)
+		to_chat(src, span_userdanger("Your shield breaks!"))
+		to_chat(src, span_brass("You require a <b>welding tool</b> to repair your damaged shield!"))
+
+
+/mob/living/basic/clockwork_marauder/welder_act(mob/living/user, obj/item/tool)
+	if(!do_after(user, 2.5 SECONDS, target = src))
```
Comment (MrMelbert) [inline modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm]: use `use_tool`, not `do_after`, or you don't get the sounds + fuel use + toolspeed mod

File: modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm (hunk 1 above)
Comment (Zonespace27) [reply modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm]: Done

File: modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm (hunk 2 above)
Comment (Zonespace27) [reply modular_skyrat/modules/clock_cult/code/mobs/clockwork_marauder.dm]: Done

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**